src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -513,6 +517,7 @@ public object GetRealObject(StreamingContext context)
 
 		public ISessionBuilder WithOptions()
 		{
+			CheckNotDisposed();
 			return new SessionBuilderImpl(this);
```

Comment:
I have mostly protected accesses to resources destroyed or disposed by the session factory closing, except here and in stateless equivalent, where there is no direct accesses to these resources. But that is the main usage of session factories, so, better to directly throw here when using it on a disposed factory.